### ISSUE TITLE
Some questions about the baseline performance

### ISSUE THREAD

**Issue by LinglanZhao** [ShuoYang-1998/ICLR2021-Oral_Distribution_Calibration#3]
Hi, thank you for your transparent and impressive work！
I tested your proposed method on mini-imagenet dataset using the provided codes, and added several lines of code to re-implement the baseline methods which only uses few support samples. I found that the simple baseline method can perform on par with (in some case, even better than) the proposed one using the same 2000 sampled testing tasks. That is surprising. The detailed results are as follows (L2 normalize denotes using l2 normalization on the provided features):
| Method  | finetune | hallucination | beta | L2 normalize | 1-shot Acc. | 5-shot Acc. |
| ------------- | ------------- |  ------------- |  ------------- |  ------------- |  ------------- |  ------------- |
| DC (proposed) | w/ |  w/  |  0.5  | w/o  |  **68.21**  |  83.46  |
| DC (proposed) | w/ |  w/  |  0.5  | w/  |  67.33  |  83.26  | 
| baseline-finetune | w/ |  w/o  |  0.5  | w/o  |  64.63  |  83.11  |
| baseline-finetune | w/ |  w/o  | 1  | w/o  |  59.07  |  81.48  |
| baseline-finetune | w/ |  w/o  |  0.5  | w/  |  65.18  |  **84.11**  |
| baseline-finetune | w/ |  w/o  | 1  | w/  |  66.39  |  83.56  |  

You can reproduce the exactly the same results using `sh eval.sh`  and all the codes that I modified are in the attachment.  It seems that the proposed method performs better than the baseline method by 2% in 1-shot case but performs inferior to the simple baseline method using only few support samples without calibration in 5-shot case. That's a surprising result!
If I made any mistake, please corret me.
[test_DC.zip](https://github.com/ShuoYang-1998/ICLR2021-Oral_Distribution_Calibration/files/5829153/test_DC.zip)

**Comment by ShuoYang-1998** [comment]
Hi linglan,

The Tukey's transformation (with beta=0.5), which calibrates the skewed distribution into a more Gaussian one, is also an important component of our key idea to calibrate the few-shot class distribution. You can find more details about the ablation study in Table.4 in the paper.

**Comment by LinglanZhao** [comment]
Thanks for your reply! @ShuoYang-1998 
However the value of beta is not the key component of my question. My question is the baseline performance (denoted as *baseline-finetune* in the table) compared to the proposed one (denoted as *DC (proposed)*). I found the *baseline-finetune* model **using only few support samples (i.e. 5/25 in 5-way 1/5-shot) without hallucinating additional features and without calibration performs on par or even better than the proposed method**. For example, *baseline-finetune* performs (66.39%) only 2% lower the DC (68.21%) in 1-shot setting. Moreover, *baseline-finetune* performs  (84.11%) even higher compred to the propsed DC (83.46%) in 5-shot setting. Note that exactly the same 2000 tasks were sampled in each setting for a fair comparison. 
That is a confusing finding! 
I also provide all the modified codes in the attachment to check if there are some mistakes in my re-implementation.  Maybe you can also provide the baseline testing code, since the baseline result in Table.4 is much lower than my re-implementation?

**Comment by ShuoYang-1998** [comment]
Your baseline is achieved by applying additional L2 Norm. However, we didn't use the L2 Norm in the paper. In that case, we improve the baseline by a large margin. Your re-implementation achieves a higher baseline performance by applying the L2 norm, which is interesting. Why our method even performs worse when applying the L2 norm is also an interesting problem. What is the gap and how to deal with this phenomenon？ Maybe you can improve our method in that aspect.

**Comment by BeierZhu** [comment]
> Your baseline is achieved by applying additional L2 Norm. However, we didn't use the L2 Norm in the paper. In that case, we improve the baseline by a large margin. Your re-implementation achieves a higher baseline performance by applying the L2 norm, which is interesting. Why our method even performs worse when applying the L2 norm is also an interesting problem. What is the gap and how to deal with this phenomenon？ Maybe you can improve our method in that aspect.

hello, thank you for your impressive work！But I think your code evaluate_DC.py uses sklearn LogisticRegression whose default penalty is L2 norm which means you actually use the L2 norm in the paper. Is my understanding right?
